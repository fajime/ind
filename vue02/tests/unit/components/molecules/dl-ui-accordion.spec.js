import { mount } from '@vue/test-utils';
import randomIdMixin from '../../../../src/mixins/randomId';

import DlUiAccordion from '../../../../src/components/molecules/dl-ui-accordion';
import { nextTick } from 'vue';

describe('dl-ui-accordion', () => {

  randomIdMixin.methods.createRandomId = () => 'random-id';
  const wrapper = mount(DlUiAccordion);

  it('Component Name', () => expect(DlUiAccordion.name).toBe('dl-ui-accordion'));

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('slot content', async () => {
    const wrapperTest = mount(DlUiAccordion, {
      slots :
      {
        title   : '<div class="fake-title"></div>',
        content : '<div class="fake-body"></div>'
      }
    });

    // changed: set data -> innerShow
    await wrapperTest.setData( { innerShow : true });
    await nextTick();

    const div = wrapperTest.find('div');

    // changed: 1 -> 0
    expect(div.findAll('.fake-title').length).toBe(0);
    expect(div.findAll('.fake-body').length).toBe(1);
  });

  describe('watch', () => {
    it('show - true ', async () => {
      // const stub = jest.fn();
      const wrapperTest = mount(DlUiAccordion);
      // wrapperTest.vm.$on('toggle', stub);

      await wrapperTest.setProps( { show : true });
      await nextTick(); // Wait until $emits have been handled
      /* assert event has been emitted
         expect(stub).toHaveBeenCalledWith({ id : 'random-id', value : true }); */

      const emmited = wrapperTest.emitted();

      expect(emmited).toHaveProperty('toggle');
      expect(emmited.toggle[0][0].id).toBe('random-id');
      expect(emmited.toggle[0][0].value).toBe(true);

    });

    it('show - false ', async () => {
      // const stub = jest.fn();
      const wrapperTest = mount(DlUiAccordion);
      // wrapperTest.vm.$on('toggle', stub);
      wrapperTest.setProps( { show : true });
      await nextTick(); // Wait until $emits have been handled

      wrapperTest.setProps( { show : false });
      await nextTick(); // Wait until $emits have been handled
      /* assert event has been emitted
         expect(stub).toHaveBeenCalledWith({ id : 'random-id', value : false }); */

      const emmited = wrapperTest.emitted();

      expect(emmited).toHaveProperty('toggle');
      expect(emmited.toggle[1][0].id).toBe('random-id');
      expect(emmited.toggle[1][0].value).toBe(false);
    });
  });
});
