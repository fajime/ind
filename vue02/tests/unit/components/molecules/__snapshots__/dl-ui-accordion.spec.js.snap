// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dl-ui-accordion renders correctly 1`] = `
<div
  class="dl-ui-accordion--hidden dl-ui-accordion"
>
  <input
    class="dl-ui-accordion__input"
    id="random-id"
    type="checkbox"
  />
  <label
    class="dl-ui-accordion__header"
    for="random-id"
  >
    <!--v-if-->
    <div
      class="dl-ui-accordion__text"
    >
      <div
        class="dl-ui-accordion--highlight dl-ui-accordion__text-title"
      >
        Título
      </div>
      <div
        class="dl-ui-accordion___text-description"
      >
        Descripción
      </div>
    </div>
    <div
      class="dl-ui-icon dl-ui-accordion__trigger-icon"
    >
      <svg>
        <!--v-if-->
        <!--v-if-->
        <use
          xlink:href="#dl-ids-icon-arrows-down-single-no_circle-outlined"
        />
      </svg>
    </div>
  </label>
  <div
    class="dl-ui-accordion__box--initial0 dl-ui-accordion__box"
    style="--accordion-max-height: 0px;"
  >
    <div
      class="dl-ui-accordion__content"
    >
      
      Lorem ipsum
      
    </div>
  </div>
</div>
`;
